application/src/main/java/org/opentripplanner/street/model/edge/StreetEdge.java

```diff
@@ -976,6 +977,29 @@ static int defaultMillimeterLength(LineString geometry) {
     return (int) (SphericalDistanceLibrary.length(geometry) * 1000);
   }
 
+  private State makeStateAfterHavingExitedNoDropOffZoneWhenReverseSearching(
+    State s0,
+    String network
+  ) {
+    var preferences = s0.getRequest().preferences();
+    var edit = doTraverse(s0, TraverseMode.WALK, false);
+    if (edit != null) {
+      edit.dropFloatingVehicle(s0.vehicleRentalFormFactor(), network, s0.getRequest().arriveBy());
+      if (network != null) {
+        edit.resetStartedInNoDropOffZone();
+      }
+      State state = edit.makeState();
+      if (state != null && network != null) {
+        var allowedNetworks = preferences.rental(state.currentMode()).allowedNetworks();
```

Comment:
Sorry to be so picky but here every millisecond counts. Can you compute this just once per edge and pass it into the method.

Reply:
What if you haven't defined any allowed networks? And what if you banned some?

Reply:
> Sorry to be so picky but here every millisecond counts. Can you compute this just once per edge and pass it into the method.

I would really like to, but for the preferences.rental(...) call I want currentMode after the logic of BikeWalkableEdge.createEditor which handles the stopping/starting walking a bike logic, and it's potentially spread across different classes because it's inheritable. If I added a clone() and currentMode() method directly to StateEditor, I could do it, by doing the StateEditor edit = doTraverse(s0, TraverseMode.WALK, false); bit just once, and then cloning a new editor for each makeStateAfterHavingExitedNoDropOffZoneWhenReverseSearching.

Reply:
Except that now that writing this comment forced me into looking at the code, I think in this specific case of mode = WALKING and bicycleWalking = false, none of the special code in createEditor is ever run, at least with the implementation we have now. So I can optimize that away.

Reply:
What I could do is get the allowed networks for both BICYCLE and SCOOTER, pass them both into makeStateAfter... and then choose which one to use after the state has been constructed. I'll try that. But I'll have to measure if it's any faster after all that.

Reply:
It's not going to make much of a difference either way, I'm afraid. I'll lift getting RoutingPreferences out of the repeatedly called function, but .rental(), State.currentMode() and .allowedNetworks/.bannedNetworks are all just getters which have zero additional code, and the alternative version would have four new function parameters, allowedBikeNetworks, bannedBikeNetwoks, allowedScooterNetworks, and bannedScooterNetworks. At that point it would look like it was in need of a container object, and RoutingPreferences already is that container object.